src/state/axisSelectors.ts

```diff
@@ -16,6 +17,62 @@ const unknownScale: ParsedScaleReturn = {
   realScaleType: undefined,
 };
 
+export function getDefaultDomainByAxisType(axisType: 'number' | string) {
+  return axisType === 'number' ? [0, 'auto'] : undefined;
+}
+
+function nilPredicate(filterNil: boolean) {
+  return function predicate<T>(item: T): boolean {
+    if (filterNil) {
+      return !isNil(item);
+    }
+    return true;
+  };
+}
+
+function makeUniq<T>(arr: ReadonlyArray<T>, allowDuplicates: boolean): ReadonlyArray<T> {
+  if (allowDuplicates) {
+    return arr;
+  }
+  return Array.from(new Set(arr));
+}
+
+function onlyAllowNumbersAndStringsAndDates<T>(item: T): string | number | Date {
+  if ((typeof item === 'number' && !Number.isNaN(item)) || typeof item === 'string' || item instanceof Date) {
+    return item;
+  }
+  return '';
+}
+
+export const selectDomainOfDataByKey: (
+  state: RechartsRootState,
+  axisType: AxisType,
+  axisId: AxisId,
+) => NumberDomain | CategoricalDomain = createSelector(
+  selectChartDataWithIndexes,
+  selectAllGraphicalItemsData,
+  selectAxisSettings,
+  (chartData, graphicalItemsData, axisSettings: AxisSettings): NumberDomain | CategoricalDomain => {
+    if (axisSettings == null) {
+      return undefined;
+    }
+    // Array.prototype.flat has good support everywhere except IE11: https://caniuse.com/?search=flat
+    const allDataSquished = graphicalItemsData
+      .flat(1)
+      .concat(chartData.chartData)
+      .map(entry => getValueByDataKey(entry, axisSettings.dataKey));
+
+    if (axisSettings.type === 'number') {
+      const onlyNumbers = allDataSquished.map(Number).filter(n => Number.isNaN(n) === false);
```

Comment:
Doesn't isNaN only check for the actual value NaN? I think this filter would include strings, etc?

Reply:
I will add a test